playbooks/postgres_tuning.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: PostgreSQL production tuning
+  hosts: ofn_servers
+  remote_user: "{{ user }}"
+
+  tasks:
+    - name: Find current postgres version
+      shell: psql postgres -c 'SHOW config_file;' | grep etc | egrep -o '[0-9]{1,}\.*[0-9]{1,}'
+      become: yes
+      become_user: postgres
+      register: pg_current_version
+
+    - debug:
+        msg: "Postgres current version: {{ pg_current_version.stdout }}"
+
+    - name: Create postgres conf.d directory
+      file:
+        dest: "/etc/postgresql/{{ pg_current_version.stdout }}/main/conf.d"
+        state: directory
+      become: yes
+
+    - name: Include conf.d directory in postgresql.conf
+      lineinfile:
+        path: "/etc/postgresql/{{ pg_current_version.stdout }}/main/postgresql.conf"
+        line: "include_dir 'conf.d'"
+      become: yes
+
+    - name: Add tuning configuration
+      copy:
+        content: |
+          shared_buffers = {{ ( ansible_memtotal_mb / 4 ) | int }}MB
+          effective_cache_size = {{ ( ( ansible_memtotal_mb / 4 ) * 3 ) | int }}MB
+          maintenance_work_mem = {{ ( ansible_memtotal_mb / 16 ) | int }}MB
+          wal_buffers = 16MB
+          min_wal_size = 1GB
+          max_wal_size = 2GB
```

Comment:
Can we make at least these two relative to `ansible_memtotal_mb` as well? We're working on getting a production server with the standard specs at Katuma but we're not there yet. 

I checked pgtune and still with a 2 GB RAM it sets max_wal_size = 2GB :man_shrugging:

Reply:
`min_wal_size` and `max_wal_size` have been removed, so they will stay at their defaults. :+1: